project/Subject.cs

```diff
@@ -1,152 +1,169 @@
-                    throw new ArgumentException("Subject name cannot be null or empty.");
-                _subjectName = value;
-            }
+            if (string.IsNullOrWhiteSpace(value))
+                throw new ArgumentException("Subject name cannot be null or empty.");
+            _subjectName = value;
         }
+    }
 
-        private int _gradingScale;
-        public int GradingScale
+    private int _gradingScale;
+    public int GradingScale
+    {
+        get => _gradingScale;
+        private set
         {
-            get => _gradingScale;
-            private set
-            {
-                if (value < 1 || value > 5)
-                    throw new ArgumentException("Grading scale must be between 1 and 5.");
-                _gradingScale = value;
-            }
+            if (value < 1 || value > 5)
+                throw new ArgumentException("Grading scale must be between 1 and 5.");
+            _gradingScale = value;
         }
+    }
 
-        private List<Teacher> _teachers = new();
-        public List<Teacher> Teachers
+    private List<Teacher> _teachers = new();
+    public List<Teacher> Teachers
+    {
+        get => new(_teachers);
+        set
         {
-            get => new(_teachers);
-            set
-            {
-                if (value == null || value.Count == 0)
-                    throw new InvalidOperationException(
-                        "A subject must be assigned to at least one teacher."
-                    );
-                _teachers = value;
-            }
+            if (value == null || value.Count == 0)
+                throw new InvalidOperationException(
+                    "A subject must be assigned to at least one teacher."
+                );
+            _teachers = value;
         }
+    }
 
-        private List<Student> _students = new();
-        private Subject _subSubject;
-        private List<Subject> _subSubjects => throw new NotImplementedException();
+    private List<Student> _students = new();
+    private Subject _subSubject;
+    private List<Subject> _subSubjects => throw new NotImplementedException();
 
-        private List<Timeslot> _timeslots => throw new NotImplementedException();
+    private List<Timeslot> _timeslots => throw new NotImplementedException();
 
-        public Subject SubSubject
+    private Teacher _teacher;
+    public Teacher Teacher
+    {
+        get => _teacher;
+        set
         {
-            get => _subSubject;
-            private set { _subSubject = value; }
-        }
+            if (value == null)
+                throw new ArgumentNullException(nameof(value), "Teacher cannot be null.");
 
-        public List<Student> Students
-        {
-            get => new(_students);
-            set
+            if (_teacher != null && _teacher != value)
+                throw new InvalidOperationException("This subject is already assigned to a different teacher.");
```

Comment:
These validation properties must not be in here, they need to be in the CRUD operations. Must be `private set` as well.